appinsights/bond.go

```diff
@@ -59,25 +53,63 @@ const (
 	Critical
 )
 
-type messageData struct {
-	domain
-	Message       string        `json:"message"`
-	SeverityLevel SeverityLevel `json:"severityLevel"`
+type MessageData struct {
+	Ver           int               `json:"ver"`
+	Properties    map[string]string `json:"properties"`
+	Message       string            `json:"message"`
+	SeverityLevel SeverityLevel     `json:"severityLevel"`
 }
 
-type requestData struct {
-	domain
+type RequestData struct {
+	Ver          int                `json:"ver"`
+	Properties   map[string]string  `json:"properties"`
 	Id           string             `json:"id"`
 	Name         string             `json:"name"`
 	StartTime    string             `json:"startTime"` // yyyy-mm-ddThh:mm:ss.fffffff-hh:mm
 	Duration     string             `json:"duration"`  // d:hh:mm:ss.fffffff
 	ResponseCode string             `json:"responseCode"`
 	Success      bool               `json:"success"`
-	httpMethod   string             `json:"httpMethod"`
-	url          string             `json:"url"`
+	HttpMethod   string             `json:"httpMethod"`
+	Url          string             `json:"url"`
 	Measurements map[string]float32 `json:"measurements"`
 }
 
+type DependencyKind int
+
+const (
+	SQL DependencyKind = iota
+	Http
+	Other
+)
+
+type DependencySourceType int
+
+const (
+	Undefined DependencySourceType = iota
+	Aic
+	Apmc
+)
+
+type RemoteDependencyData struct {
+	Ver              int                  `json:"ver"`
+	Id               string               `json:"id"`
+	Name             string               `json:"name"`
+	ResultCode       int                  `json:"resultCode"`
+	CommandName      string               `json:"commandName"`
+	Kind             DataPointType        `json:"kind"`
+	Duration         string               `json:"duration"`
+	Count            int                  `json:"count"`
+	Min              float32              `json:"min"`
+	Max              float32              `json:"max"`
+	StdDev           float32              `json:"stdDev"`
```

Comment:
I'm not entirely sure this is the correct data.  At least, it is not what I see when I compare it to the data found in the [.Net sdk](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/src/Core/Managed/Shared/Extensibility/Implementation/External/RemoteDependencyData_types.cs).  I wonder if @SergeyKanzhelev or @AlexBulankou can comment.